test/e2e/autoscale_test.go

```diff
@@ -357,12 +383,90 @@ func assertAutoscaleUpToNumPods(ctx *testContext, curPods, targetPods float64, d
 	}
 }
 
+func assertGracefulScaledown(t *testing.T, ctx *testContext, size int) error {
+	// start x running pods; x == size
+	hostConnMap, err := uniqueHostConnections(t, ctx.names, size)
+	if err != nil {
+		return err
+	}
+
+	// only keep openConnCount connections open for the test
+	openConnCount := size / 2
+	deleteHostConnections(hostConnMap, size-openConnCount)
+
+	defer deleteHostConnections(hostConnMap, openConnCount)
+	grp := errgroup.Group{}
+	grp.Go(func() error {
```

Comment:
Why make this a group and async in the first place? You're only launching a single goroutine and then you immediately wait for it to finish, soooo.... Could it just be synchronous code instead?

Reply:
I had it as a single process initially but split the process for consistency in shape to other tests. can change it back.